src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/network/access/UnableAccessProvider.kt

```diff
@@ -0,0 +1,13 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.network.access
+
+import com.amazonaws.services.ec2.model.SecurityGroup
+
+class UnableAccessProvider : AccessProvider {
```

Comment:
How about renaming it to NoAccessProvider? Unless I'm getting the intent wrong here.

Reply:
Not sure if it wouldn't be another implementation based on the `@return` value definition I made for `AccessProvider#provideAccess()`. I mean one that would always return `true` instead of `false`, because providing an access to nothing is always certain success.

The `UnableAccessProvider` is essentially linked to that definition of the `@return`. It can't provide you any access, because it's "unable". We may change both the class name and the returned value. Let me know if you have any suggestions about the latter.

The current `@return` definition:
> @return true if it was possible to provide access, false otherwise

Reply:
- [ ] Change return type of `UnableAccessProvider` to 3 value logic

Reply:
A while ago I discussed the topic of values returned by `AccessProvider`/`AccessRequester` with @dagguh. We agreed on changing it to 3 value logic to model service-like response (like http 200/400/500 response codes), however now when I'm trying to implement it I again think it should be left as simple as it is. I don't want to model exact behavior of the implementation underneath this interface. The caller should not be interested in recovering from any failures - it's up to the implementation to provide as reliable logic as possible. The problem is that we are not injecting dependencies (i.a. requester factory) into the domain level logic. For it to be done properly this code would need to be restructured from ground up and I believe we don't have time for that.

I'm fine with changing the name to `NoAccessProvider` and making this class return `true` to indicate that the task of not granting any access was successful. I could then also change the definition of returned value to `true if access grant was successful, false otherwise`, as it may be more intuitive than `true if it was possible to grant access, false otherwise`. 

@dagguh, @mzyromski-atlassian, @mgrzaslewicz what do you think about it?

Reply:
+1 for renaming to `NoAccessProvider` and following the `true if it was possible to grant access, false otherwise` convention.

Reply:
- [ ] Change the `AccessProvider` return value doc
- [ ] Rename `UnableAccessProvider` to `NoAccessProvider`